Dalamud/Game/Addon/Lifecycle/AddonEvent.cs

```diff
@@ -1,72 +1,121 @@
-﻿namespace Dalamud.Game.Addon.Lifecycle;
+﻿using Dalamud.Game.Addon.Lifecycle.AddonArgTypes;
+
+using FFXIVClientStructs.FFXIV.Component.GUI;
+
+namespace Dalamud.Game.Addon.Lifecycle;
 
 /// <summary>
 /// Enumeration for available AddonLifecycle events.
 /// </summary>
 public enum AddonEvent
 {
     /// <summary>
-    /// Event that is fired before an addon begins it's setup process.
+    /// An event that is fired prior to an addon being setup with its implementation of
+    /// <see cref="AtkUnitBase.OnSetup"/>. This event is useful for modifying the initial data contained within
+    /// <see cref="AddonSetupArgs.AtkValueSpan"/> prior to the addon being created.
     /// </summary>
+    /// <seealso cref="AddonSetupArgs"/>
     PreSetup,
-
+    
     /// <summary>
-    /// Event that is fired after an addon has completed it's setup process.
+    /// An event that is fired after an addon has finished its initial setup. This event is particularly useful for
+    /// developers seeking to add custom elements to now-initialized and populated node lists, as well as reading data
+    /// placed in the AtkValues by the game during the setup process.
+    /// See <see cref="PreSetup"/> for more information.
     /// </summary>
     PostSetup,
 
     /// <summary>
-    /// Event that is fired before an addon begins update.
+    /// An event that is fired before an addon begins its update cycle via <see cref="AtkUnitBase.Update"/>. This event
+    /// is fired every frame that an update is loaded, regardless of visibility.
```

Comment:
I think the wording is a bit weird here.
```suggestion
    /// An event that is fired before an addon begins its update cycle via <see cref="AtkUnitBase.Update"/>. This event
    /// is fired every frame, regardless of visibility.
```